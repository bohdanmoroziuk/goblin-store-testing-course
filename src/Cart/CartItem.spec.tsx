import { fireEvent } from '@testing-library/react';
import React from 'react'
import { Product } from 'shared/types'
import { CartItem } from './CartItem';

describe("CartItem", () => {
  const product: Product = {
    name: "Foo",
    price: 55,
    image: "/images/basic_spear.png"
  };

  const mockRemoveFromCart = jest.fn();

  it("renders correctly", () => {
    const { container, getByAltText } = renderWithRouter(
      () => <CartItem product={product} removeFromCart={mockRemoveFromCart} />
    );

    expect(container.innerHTML).toMatch(product.name);
    expect(container.innerHTML).toMatch(product.price.toString());
    expect(getByAltText(product.name)).toHaveAttribute("src", product.image);
    expect(container).toMatchSnapshot();
  })

  describe("on 'Remove item' click", () => {
    it("removes the item from the cart", () => {
      const { getByText } = renderWithRouter(
        () => <CartItem product={product} removeFromCart={mockRemoveFromCart} />
      );

      fireEvent.click(getByText("Remove"));

      expect(mockRemoveFromCart).toBeCalledWith(product);
    })
  })
})