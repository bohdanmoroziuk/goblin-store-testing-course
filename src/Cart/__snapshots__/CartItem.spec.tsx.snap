// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly 1`] = `
<div>
  <div
    class="cart-item"
  >
    <img
      alt="Foo"
      height="64px"
      src="/images/basic_spear.png"
      style="image-rendering: pixelated;"
      width="64px"
    />
    <p>
      Foo
    </p>
    <p>
      55
       Zm
    </p>
    <button
      class="nes-btn is-error"
    >
      Remove
    </button>
  </div>
</div>
`;
